src/features/zaaksysteem/service.ts

```diff
@@ -28,14 +28,18 @@ const combineOverview = (multiple: ZaakDetails[][]) =>
 
 const fetchZaakOverview = (systeem: Systeem, query: URLSearchParams) =>
   fetchWithSysteemId(systeem.identifier, `${zaaksysteemBaseUri}/zaken?${query}`)
-    .then(throwIfNotOk)
     .then(parseJson)
     .then((json) =>
       parsePagination(json, (x) =>
         mapZaakDetails({ ...(x as any), zaaksysteemId: systeem.identifier }),
       ),
     )
-    .then(({ page }) => page);
+    .then(({ page }) => page)
+    .catch(() => {
```

Comment:
I think what happens now is:
- we assume every response from the zaken endpoint contains json, even when the status code is not ok
- when the response is not json or the json doesn't match the pagination structure we expect, there is probably an exception
- we catch any exception and return null in that case

I think we can be more specific:
- we only expect this for queries by niet-natuurlijk-personen
- we only expect 400 responses if the query is not supported

The function that was used before, `throwIfNotOk`, throws an Error with the statusCodeText if the response is not ok. We could catch that from the fetchZakenByKlantBedrijfIdentifier function and check if it is '400' there. What do you think?